### PR TITLE
GSNGM-717: Download data in an area: New tool and add more data sets (add download the entire layers)

### PR DESCRIPTION
**Pull request by b4l**

### CODE REVIEW

**Comment by eleu** [comment]
Nice!

While testing, I found the following issues:
* Can we grey out the download button of the Faults / Boreholes if we are in the case "No data to download"?
* We do not necessary have to zip the boreholes. If Zip is needed for technical reasons, please remove 2 folder level inside the zip file ("swissgeol-data" and "boreholes" folders are not needed).
* If I have drawn a rectangle, make that one active (= show the list of downloadable layers).
* I have the impression that the scroll down to the just drawn geom does not work correctly. Sometimes it scrolls down half way, don't know exactly in which cases.

I also found an issue for the "Download" button in the floating geom info panel (does only open tools choice on first click), and besides that some usability inconsistencies (button should probably be greyed out if the tool is already open, or named differently, or integrate the downloadable list directly in the floating panel...). I think it would be better to just remove this button for the moment.

**Comment by b4l** [comment]
>     * Can we grey out the download button of the Faults / Boreholes if we are in the case "No data to download"?

We can but it is probably not worth the hassle. We wold need to download the data first and keep track on editing changes.

>     * We do not necessary have to zip the boreholes. If Zip is needed for technical reasons, please remove 2 folder level inside the zip file ("swissgeol-data" and "boreholes" folders are not needed).

Done!
 
>     * If I have drawn a rectangle, make that one active (= show the list of downloadable layers).

I am not sure whats the best way to do this. Maybe someone has a suggestion.

>     * I have the impression that the scroll down to the just drawn geom does not work correctly. Sometimes it scrolls down half way, don't know exactly in which cases.

Same as above
 
> I also found an issue for the "Download" button in the floating geom info panel (does only open tools choice on first click), and besides that some usability inconsistencies (button should probably be greyed out if the tool is already open, or named differently, or integrate the downloadable list directly in the floating panel...). I think it would be better to just remove this button for the moment.

Removed it!

File: src/toolbox/ngm-toolbox.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ import {showSnackbarInfo} from '../notifications';
 @customElement('ngm-tools')
 export class NgmToolbox extends LitElementI18n {
   @property({type: Boolean}) toolsHidden = true;
-  @state() activeTool: 'draw' | 'slicing' | 'gst' | undefined;
+  @property() activeTool: 'draw' | 'slicing' | 'gst' | 'data-download' | undefined;
```
Comment (vladyslav-tk) [inline src/toolbox/ngm-toolbox.ts:30]: why `@property`?

**Review by vladyslav-tk — APPROVED** [review]

**Comment by vladyslav-tk** [comment]
@b4l 
> * If I have drawn a rectangle, make that one active (= show the list of downloadable layers).
> I am not sure whats the best way to do this. Maybe someone has a suggestion.

I made it

@eleu 
> * I have the impression that the scroll down to the just drawn geom does not work correctly. Sometimes it scrolls down half way, don't know exactly in which cases.

I can't reproduce it

**Comment by b4l** [comment]
@vladyslav-tk thanks! I reverted the properties to states. It was necessary to open and select the geometry when clicking the download button in the floating geometry info but the button is removed now.

File: src/elements/ngm-side-bar.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export class SideBar extends LitElementI18n {
   @property({type: Boolean}) mobileView = false;
   @state() catalogLayers: any;
   @state() activeLayers: any;
-  @state() activePanel: string | null = null;
+  @property() activePanel: string | null = null;
```
[the remark this answers is not in the document]
Comment (b4l) [reply src/elements/ngm-side-bar.ts]: reverted in https://github.com/swissgeol/ngm/pull/773/commits/fdadeb903bf1ac1decea4be8ded6be9bdbd6a591

File: src/toolbox/data-download.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+import type {CustomDataSource} from 'cesium';
+import type {NgmGeometry} from './interfaces';
+import type {Config} from '../layers/ngm-layers-item';
+
+
+@customElement('data-download')
+export class DataDownload extends LitElementI18n {
+  @property({type: Object}) geometriesDataSource: CustomDataSource | undefined;
+  @property({type: String}) selectedGeometryId = '';
```
[the remark this answers is not in the document]
Comment (b4l) [reply src/toolbox/data-download.ts]: reverted in https://github.com/swissgeol/ngm/pull/773/commits/fdadeb903bf1ac1decea4be8ded6be9bdbd6a591

File: src/toolbox/ngm-toolbox.ts (hunk 1 above)
Comment (b4l) [reply src/toolbox/ngm-toolbox.ts]: reverted in https://github.com/swissgeol/ngm/pull/773/commits/fdadeb903bf1ac1decea4be8ded6be9bdbd6a591

**Pull request merged**